kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/data/vector/DefaultArrayVector.java

```diff
@@ -64,19 +63,26 @@ public DefaultArrayVector(
      * @return
      */
     @Override
-    public <T> List<T> getArray(int rowId) {
+    public ArrayValue getArray(int rowId) {
```

Comment:
I'd add a comment explaining what's going on here:

We have some `offsets`. Then we use these offsets to get a start and end. Then we create a ViewVector ... this is all non-trivial. Explain how / why.